src/lib.rs

```diff
@@ -270,8 +270,20 @@ fn open_browser_internal(browser: Browser, url: &str) -> Result<ExitStatus> {
     match browser {
         Browser::Default => open_on_unix_using_browser_env(url)
             .or_else(|_| -> Result<ExitStatus> { Command::new("xdg-open").arg(url).status() })
+            .or_else(|r| -> Result<ExitStatus> {
+                if let Ok(desktop) = ::std::env::var("XDG_CURRENT_DESKTOP") {
+                    if desktop == "KDE" {
+                        return Command::new("kioclient").arg("exec").arg(url).status();
+                    }
+                }
+                Err(r) // If either `if` check fails, fall through to the next or_else
+            })
             .or_else(|_| -> Result<ExitStatus> { Command::new("gvfs-open").arg(url).status() })
-            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() }),
+            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() })
+            .or_else(|_| -> Result<ExitStatus> {
+                Command::new("kioclient").arg("exec").arg(url).status()
+            })
+            .or_else(|_| -> Result<ExitStatus> { Command::new("x-www-browser").arg(url).status() }),
```

Comment:
I'll recommend you modify this to something like
```rust
.or_else(|e| -> Result<ExitStatus> {
                if let Ok(child) = Command::new("x-www-browser").arg(url).spawn() {
                    return Ok(ExitStatusExt::from_raw(0));
                }
                Err(e)
            })
```
to allow for the tests to pass for now. It'll allow me to accept the PR. I'm realising that a larger fix for making the behaviour consistent, and reflecting those in the tests, will take me some time. So let's focus on closing this out for now.

Reply:
OK. I'm just winding down for bed but I should have a commit for you tomorrow.

Reply:
I pushed the change but it's still blocking on the browser.

I have a headache and want to get to bed early, so I'll have to look into it further tomorrow, but I know it's not `kioclient exec` because, if I temporarily set my KDE handler for `http://` and `https://` URLs to something that's not already running and, `kioclient exec` will launch the expected browser but will not block.

That leaves two possibilities I can think of off the top of my head:

1. The code snippet you proposed is insufficient to keep it from waiting on a spawned child. I'll have to experiment tomorrow to rule this out.
2. Travis-CI's configuration has changed since you last ran a test and, if you were to ask it to re-test your current `HEAD`, that would timeout and fail too.

I'm fairly certain it's not a difference between how my and Travis-CI's copies of commands behave, because we're both running systems based on Ubuntu Xenial (16.04 LTS).

Reply:
I ran a [new build](https://travis-ci.com/amodm/webbrowser-rs/builds/124193251) on HEAD just now, and it passed.